### PR TITLE
fix: API unit test to use only the plugin and not Amplify

### PR DESCRIPTION
**Pull request by royjit**
*Description of changes:*

Unit test of API plugin is currently required to configure Amplify. This PR removes that dependency and makes the unit test light as possible and test the unit of code.

*Check points:*

- [ ] ~Added new tests to cover change, if needed~
- [x] All unit tests pass
- [ ] All integration tests pass
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/OperationTestBase.swift [hunk 1]
```diff
@@ -12,21 +12,23 @@ import XCTest
...
+    var apiPlugin: AWSAPIPlugin!
+
     /// There are no throwing methods in this, but in order to let subclasses have a predictable way
     /// of setting up plugins, this base class uses `setUpWithError` instead of `setUp`. (XCTest
     /// lifecycle normally invokes `setUp` after `setUpWithError`, which means that any state config
     /// done inside of a subclass' `setUpWithError` could be erased by this class' call to
     /// `Amplify.reset` from inside `setUp`.
     override func setUpWithError() throws {
-        Amplify.reset()
+        apiPlugin = nil
```
Comment (palpatim) [inline AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/OperationTestBase.swift:23]: Does `AWSAPIPlugin.reset()` do anything to cancel in-process operations? I'm not sure whether just setting `nil` will do enough to clear in-process/shared state

File: AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/OperationTestBase.swift [hunk 2]
```diff
@@ -12,21 +12,23 @@ import XCTest
 
 class OperationTestBase: XCTestCase {
 
+    var apiPlugin: AWSAPIPlugin!
+
     /// There are no throwing methods in this, but in order to let subclasses have a predictable way
     /// of setting up plugins, this base class uses `setUpWithError` instead of `setUp`. (XCTest
     /// lifecycle normally invokes `setUp` after `setUpWithError`, which means that any state config
     /// done inside of a subclass' `setUpWithError` could be erased by this class' call to
     /// `Amplify.reset` from inside `setUp`.
```
Comment (palpatim) [inline AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/OperationTestBase.swift:21]: Make sure to adjust comment to reflect whatever our final state is

Comment (royjit) [reply AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/OperationTestBase.swift]: Fixed

File: AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/OperationTestBase.swift (hunk 1 above)
Comment (royjit) [reply AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/OperationTestBase.swift:30]: Added reset for plugin

**Review by palpatim — APPROVED** [review]

**Pull request merged**